Project/Assets/Editor/Twine Importer/TwineJsonParser.cs

```diff
@@ -2,167 +2,240 @@
+        ActivateVariableExpressions(variableExpressions, twineNode);
 
 		// Upon creation of this node, ensure that it is a decision node if it has
 		//	the decision tag:
+		// Vice versa for condition node
 		twineNode.isDecisionNode = (twineNode.tags != null && twineNode.tags.Contains (PRAIRIE_DECISION_TAG));
+		twineNode.isConditionNode = (twineNode.tags != null && twineNode.tags.Contains (PRAIRIE_CONDITION_TAG));
 
 		// Start all twine nodes as deactivated at first:
 		twineNode.Deactivate();
 
 		return nodeGameObject;
 
-		#endif
-	}
-
-	public static void MatchChildren(Dictionary<string, JSONNode> twineNodesJsonByName, Dictionary<string, GameObject> gameObjectsByName)
-	{
-		foreach(KeyValuePair<string, GameObject> entry in gameObjectsByName)
-		{
-			string nodeName = entry.Key;
-			GameObject nodeObject = entry.Value;
-
-			TwineNode twineNode = nodeObject.GetComponent<TwineNode> ();
-			JSONNode jsonNode = twineNodesJsonByName [nodeName];
-		
-			// Iterate through the links and establish object relationships:
-			JSONNode nodeLinks = jsonNode ["links"];
-
-			twineNode.children = new GameObject[nodeLinks.Count];
-			twineNode.childrenNames = new string[nodeLinks.Count];
-
-			for (int i = 0; i < nodeLinks.Count; i++) {
-				JSONNode link = nodeLinks [i];
-				string linkName = link ["name"];
-				GameObject linkDestination = gameObjectsByName [link ["link"]];
-
-				// Remember parent:
-				linkDestination.GetComponent<TwineNode> ().parents.Add (nodeObject);
-
-				// Set link as a child, and remember the name.
-				twineNode.children[i] = linkDestination;
-				twineNode.childrenNames [i] = linkName;
-			}
-		}
-	}
-		
-	/// <summary>
-	/// Strips the list of children off the content,
-	/// because we really only want the content.
-	/// </summary>
-	/// <returns>The content without children atached.</returns>
-	/// <param name="content">Content with children attached.</param>
-	public static string RemoveTwineLinks (string content)
-	{
-		string[] substrings = content.Split ('[');
-		return substrings[0];
-	}
-
-	static string[] GetDequotedStringArrayFromJsonArray (JSONNode jsonNode)
-	{
-		if (jsonNode == null) {
-			return null;
-		}
-
-		string[] stringArray = new string[jsonNode.Count];
-		for (int i = 0; i < jsonNode.Count; i++)
-		{
-			string quotedString = jsonNode [i].ToString();
-			string dequotedString = quotedString.Replace ('"', ' ').Trim ();
-			stringArray [i] = dequotedString;
-		}
-
-		return stringArray;
-	}
+#endif
+    }
+
+    /// <summary>
+    /// Parses the text of a twine node to find the variable expressions 
+    /// (marked with double parentheses) within
+    /// </summary>
+    /// <param name="text">Node text with all links and variable expressions</param>
+    /// <returns>Variable expressions minus parentheses</returns>
+    public static string[] GetVariableExpressions(string text)
+    {
+        Regex expressionRegex = new Regex("\\(\\([^)]*\\)\\)");
```

Comment:
Can you add some comments with examples maybe to indicate what kind of expressions are this is trying to parse? Thank you!